### PR TITLE
CB-9116 Ensure platform scripts are executable as the platform could be added on non-OSX environment

### PR DESCRIPTION
**Pull request by daserge**
Corresponding [Jira issue](https://issues.apache.org/jira/browse/CB-9116)

### CODE REVIEW

File: bin/templates/scripts/cordova/lib/build.js [hunk 1]
```diff
@@ -87,6 +87,13 @@ module.exports.run = function (argv) {
         }
         return Q.nfcall(fs.writeFile, path.join(__dirname, '..', 'build-extras.xcconfig'), extraConfig, 'utf-8');
     }).then(function () {
+        // Ensure platform scripts are executable as the platform could be added on non-OSX environment
+        shell.find(projectPath).filter(function (file) {
```
Comment (sgrebnov) [inline bin/templates/scripts/cordova/lib/build.js]: Should we check `cordova` folder only? (projectPath/cordova). My concern is that there could be too many files in other folders like `www` so this could be slow.

Comment (daserge) [inline bin/templates/scripts/cordova/lib/build.js]: Makes sense; although there is another script there: 
`ios/build/HelloCordova.build/Debug-iphonesimulator/HelloCordova.build/Script-304B58A110DAC018002A0835.sh`
So we can probably check all folders except `www` - what do you think?

**Comment by omefire** [comment]
LGTM !
- what causes file permissions to change when the platform is added on a non-OSX system ?

**Comment by daserge** [comment]
@omefire I think that it is because git does not include exec permission for these files.

**Comment by omefire** [comment]
@daserge , does this behavior show up if for example, I add android from windows, then check it into git ?

**Comment by daserge** [comment]
@omefire I don't have Android environment on Mac now, but as far as I can see there are no shell scripts in Android platform.

File: bin/templates/scripts/cordova/lib/build.js (hunk 1 above)
Comment (sgrebnov) [inline bin/templates/scripts/cordova/lib/build.js]: sounds good

**Comment by nikhilkh** [comment]
Does it make sense to convert any of these scripts to node instead of 'shell'?

**Comment by omefire** [comment]
It'd be a great idea to convert the scripts from 'shell' to 'node'.
bash is hard to follow and make sense of.

**Comment by daserge** [comment]
Closing this - decided to rewrite the scripts in node.
